engine/sceneManager/components/Component.hpp

```diff
@@ -0,0 +1,29 @@
+﻿//
+// Created by larsv on 12/11/2024.
+//
+
+#ifndef BLOCKYENGINE_COMPONENT_HPP
+#define BLOCKYENGINE_COMPONENT_HPP
+
+#include <memory>
+#include <string>
+
+#include "sceneManager/components/ComponentTransform.hpp"
+
+class GameObject;
```

Comment:
Vermijd de forward declaration hier, Google style guide geeft een best valide reden as to why